### PR TITLE
Pivotal ID # 184353778: Add gitlab pipeline to run e2e

### PR DESCRIPTION
**Pull request by msrada94**
https://www.pivotaltracker.com/story/show/184353778

List of the changes:
* Added jobs to build and run e2e tests

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+build:
+  stage: build
+  script: gradle clean build
+
+run-e2e:
+  needs:
+    - build
+  stage: tests
+  script: gradle clean test
```
Comment (Juan-EBI) [inline .gitlab-ci.yml:15]: if you clean basically you are losing everything result of build, so gradle will need to build again

**Review by Juan-EBI — CHANGES_REQUESTED** [review]
please check comment and remove code not part of this PR

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  - build
+
+build:
+  stage: build
+  script:
+    - mkdir -p ~/.docker
+    - BASE_64=$(echo -n "$CI_REGISTRY_USER:$CI_REGISTRY_PASSWORD" | base64)
+    - echo "{\"auths\":{\"$CI_REGISTRY\":{\"auth\":\"$BASE_64\"}}}" > ~/.docker/config.json
+    - BIOSTUDIES_DIR="$(mkdir ~/biostudiesMount && pwd)/biostudiesMount"
+    - gradle clean build -PbiostudiesDir=$BIOSTUDIES_DIR -PdockerUser=$CI_REGISTRY_USER -PdockerPassword=$CI_REGISTRY_PASSWORD --stacktrace
```
Comment (Juan-EBI) [inline .gitlab-ci.yml:19]: why do we need docker user and password if authentication is realdy save in `config.json`

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+stages:
+  - build
+
+build:
+  stage: build
+  script:
+    - mkdir -p ~/.docker
+    - BASE_64=$(echo -n "$CI_REGISTRY_USER:$CI_REGISTRY_PASSWORD" | base64)
+    - echo "{\"auths\":{\"$CI_REGISTRY\":{\"auth\":\"$BASE_64\"}}}" > ~/.docker/config.json
+    - BIOSTUDIES_DIR="$(mkdir ~/biostudiesMount && pwd)/biostudiesMount"
```
Comment (Juan-EBI) [inline .gitlab-ci.yml:18]: lets just call it `biostudies-workdir`

File: src/test/kotlin/listener/TestListener.kt [hunk 4]
```diff
@@ -1,44 +1,59 @@
...
+import kotlin.io.path.absolutePathString
+import kotlin.io.path.exists
 
 class TestListener : TestExecutionListener {
     override fun testPlanExecutionStarted(testPlan: TestPlan) {
-        biostudiesEnv.start()
-        ContextVariables["ENV_URL"] = "http://localhost:${biostudiesEnv.getServicePort("biostudies", 8081)}"
-        ContextVariables["ENV_FTP"] = "$biostudiesBaseFolder/ftp"
-        ContextVariables["ENV_SUBMISSION"] = "$biostudiesBaseFolder/submission"
+        runCatching { biostudiesEnv.start() }.getOrThrow()
```
Comment (Juan-EBI) [inline src/test/kotlin/listener/TestListener.kt:17]: this does nothing as it capture the exception and then throw it.

File: src/test/resources/docker/docker-compose.yml [hunk 5]
```diff
@@ -34,8 +34,8 @@ services:
       retries: 3
 
   biostudies:
-    image: dockerhub.ebi.ac.uk/jcamilorada/biostudies-backend-service/submitter:b7c1e503bc0e93356b26e63d8f8b153c9fb3e847
-    command: docker login dockerhub.ebi.ac.uk -u $USER -p $PASSWORD
+    image: dockerhub.ebi.ac.uk/jcamilorada/biostudies-backend-service/submitter:1f7ccee6d90c4357ed83d959478bf42b45f8f591-SNAPSHOT
+    command: docker login dockerhub.ebi.ac.uk -u $DOCKER_USER -p $DOCKER_PASSWORD
```
Comment (Juan-EBI) [inline src/test/resources/docker/docker-compose.yml:38]: this is not needed it. Is it?

**Review by Juan-EBI — CHANGES_REQUESTED** [review]
Looking good please check comments

File: .gitlab-ci.yml (hunk 2 above)
Comment (msrada94) [reply .gitlab-ci.yml:19]: that's true those gradle properties are not needed

File: src/test/resources/docker/docker-compose.yml (hunk 5 above)
Comment (msrada94) [reply src/test/resources/docker/docker-compose.yml]: wont be needed after this changes

File: src/test/kotlin/listener/TestListener.kt (hunk 4 above)
Comment (msrada94) [reply src/test/kotlin/listener/TestListener.kt:17]: yes it does. i got rid of this

**Review by Juan-EBI — APPROVED** [review]
LGTM

**Pull request merged**